wled00/FX_fcn.cpp

```diff
@@ -92,9 +93,15 @@ Segment::Segment(const Segment &orig) {
   _t = nullptr; // copied segment cannot be in transition
   name = nullptr;
   data = nullptr;
+  buffer1 = nullptr;
+  buffer2 = nullptr;
   _dataLen = 0;
   if (orig.name) { name = new char[strlen(orig.name)+1]; if (name) strcpy(name, orig.name); }
   if (orig.data) { if (allocateData(orig._dataLen)) memcpy(data, orig.data, orig._dataLen); }
+
+  uint16_t len = virtualLength();
```

Comment:
Although `Segment` content was copied it may be clearer to use `orig.length()`.
Do not use `virtualLength()` as its return value changes when using grouping, spacing and mirroring.

Reply:
In the latest version, I am still using `virtualLength()` but I'm dynamically growing the buffer if needed.